vertx-web/src/main/java/io/vertx/ext/web/Route.java

```diff
@@ -219,6 +242,23 @@ default Route useNormalisedPath(boolean useNormalizedPath) {
   @Fluent
   Route useNormalizedPath(boolean useNormalizedPath);
 
+  /**
+   * @return the metadata of this route, never returns null.
+   */
+  Map<String, Object> metadata();
```

Comment:
rather something more concrete like `attachments`

Reply:
@vietj there was a discussion on the issue about the naming. `attachment` can be confusing in the routing context, where there is already such method to signal that the response will have `content disposition` as `attachment`.

Other frameworks have the same method as a shorthand for setting the required http headers. We discussed that `metadata` would be the most convenient as given the use cases listed in the issue.

What still misses IMHO here is documentation as there was lots of discussions on the behavior of this feature that need to clearly documented.